examples/localstack/types.go

```diff
@@ -0,0 +1,306 @@
+	port: 4592,
+}
+
+// SecretsManager is the SecretsManager service
+var SecretsManager = Service{
+	name: "secretsmanager",
+	port: 4584,
+}
+
+// StepFunctions is the StepFunctions service
+var StepFunctions = Service{
+	name: "stepfunctions",
+	port: 4585,
+}
+
+type localStackContainerOption func(req *LocalStackContainerRequest)
+
+// WithDefaultRegion uses the default region for the container, which is "us-east-1"
+var WithDefaultRegion = WithRegion(defaultRegion)
+
+// WithCredentials returns a function that can be used to configure the AWS credentials of the container
+func WithCredentials(c Credentials) func(req *LocalStackContainerRequest) {
+	return func(req *LocalStackContainerRequest) {
+		if req.Env == nil {
+			req.Env = map[string]string{}
+		}
+
+		if c.AccessKeyID == "" {
+			c.AccessKeyID = defaultAccessKeyID
+		}
+		if c.SecretAccessKey == "" {
+			c.SecretAccessKey = defaultSecretAccessKey
+		}
+		if c.Token == "" {
+			c.Token = defaultToken
+		}
+
+		req.Env["AWS_ACCESS_KEY_ID"] = c.AccessKeyID
+		req.Env["AWS_SECRET_ACCESS_KEY"] = c.SecretAccessKey
+		req.Env["AWS_SESSION_TOKEN"] = c.Token
+	}
+}
+
+// WithRegion returns a function that can be used to configure the AWS region of the container
+func WithRegion(region string) func(req *LocalStackContainerRequest) {
+	return func(req *LocalStackContainerRequest) {
+		if req.Env == nil {
+			req.Env = map[string]string{}
+		}
+
+		if region == "" {
+			region = defaultRegion
+		}
+
+		req.Env["DEFAULT_REGION"] = region
+		req.region = region
+	}
+}
+
+// WithLegacyMode uses the legacy mode for the container, which exposes each service on a different port
+var WithLegacyMode = func(req *LocalStackContainerRequest) {
+	req.legacyMode = true
+}
+
+// WithServices returns a function that can be used to configure the container
+func WithServices(services ...Service) func(req *LocalStackContainerRequest) {
+	return func(req *LocalStackContainerRequest) {
+		serviceNames := []string{}
+		servicesPorts := map[int]bool{} // map of unique exposed ports
+
+		for _, service := range services {
+			// automatically set legacy mode for services that require it,
+			// as it will affect how the service is exposed
+			service.legacyMode = req.legacyMode
+
+			servicePort := service.servicePort()
+
+			servicesPorts[servicePort] = true
+			serviceNames = append(serviceNames, service.Name())
+
+			// add services to the list of enabled services
+			req.enabledServices = append(req.enabledServices, service)
+		}
+
+		if req.Env == nil {
+			req.Env = map[string]string{}
+		}
+
+		if len(serviceNames) > 0 {
+			req.Env["SERVICES"] = strings.Join(serviceNames, ",")
+		}
+
+		exposedPorts := []string{}
+		for port := range servicesPorts {
+			exposedPorts = append(exposedPorts, fmt.Sprintf("%d/tcp", port))
+		}
+
+		req.ExposedPorts = exposedPorts
+	}
+}
+
+// WithDefaultVersion uses the default version for the container, which is "0.11"
+var WithDefaultVersion = WithVersion(defaultVersion)
```

Comment:
IMO, given this is part of what is going to become a module, we should avoid providing default versions in the api. This is something deprecated in tc-java.

Reply:
You mean making it internal, and internally check if a version was not provided to apply the default?